akka-actor-tests/src/test/scala/akka/actor/ExtensionSpec.scala

```diff
@@ -46,12 +46,9 @@ class DummyExtensionImpl extends Extension
 
 // Dont't place inside ActorSystemSpec object, since it will not be garbage collected and reference to system remains
 class FailingTestExtension(val system: ExtendedActorSystem) extends Extension {
-  // first time the actor is created
-  val ref = system.actorOf(Props.empty, "uniqueName")
-  // but the extension initialization fails
-  // second time it will throw exception when trying to create actor with same name,
-  // but we want to see the first exception every time
-  throw new FailingTestExtension.TestException
+  // Always fail, but 'hide' this from IntelliJ to avoid compilation issues:
+  if (42.toString == "42")
+    throw new FailingTestExtension.TestException
```

Comment:
Hmmm, isn't this changing behavior, before it would pass the actorOf on first try and fail with TestException, while the second try it would fail because the `uniqueName` is taken

Reply:
Interesting, indeed it looks like that was the original intention, but we don't seem to have any test expecting that anymore?

Reply:
Or would it be testing that the `uniqueName` actor is correctly stopped when loading the extension fails?

Reply:
It kind of looks like that from the test case, but how would it be known that the extension spawned an actor? (I wrote the test and the comments but I don't remember)

Reply:
I don't understand what you think is the problem here?
The test verifies that `FailingTestExtension.TestException` is thrown when trying to access the extension both first and second time. The second time the extension mechanism should know about that the original init fails and therefore it throws the same (cached) exception again without trying to create the extensions instance again.

If it would create the extension instance again on the second access it would throw InvalidActorNameException since "uniqueName" is already taken. That would not be expected behavior and the test would fail.

Reply:
Yes, that wasn't the original case in the PR though, @raboof has since reverted a bit of the change.